tests/Http/Controllers/Views/KpiControllerTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Biigle\Modules\Kpis\Tests\Http\Controllers\Views;
+
+use ApiTestCase;
+
+class KpiControllerTest extends ApiTestCase
+{
+    public function testShow(){
+
+        $kpiPath = "admin/kpis/5";
+
+        $this->beUser();
+        $response = $this->get($kpiPath)->assertForbidden();
+
+        $this->beGuest();
+        $response = $this->get($kpiPath)->assertForbidden();
+
+        $this->beEditor();
+        $response = $this->get($kpiPath)->assertForbidden();
+
+        $this->beExpert();
+        $response = $this->get($kpiPath)->assertForbidden();
+
+        $this->beAdmin();
+        $response = $this->get($kpiPath)->assertForbidden();
+
+        $this->beGlobalReviewer();
+        $response = $this->get($kpiPath);
+        $response->assertStatus(200);
+
+        $this->beGlobalAdmin();
+        $response = $this->get($kpiPath);
+        $response->assertStatus(200);
+
+        // Indices > 5 are set to 5 (last month index)
```

Comment:
Isn't it 6 in the controller? Maybe you can perform a redirect in that case or a 404 not found? This can be tested more easily, too.